doc/performance-investigations/0009-eq-session-datastore-over-redis/summary.md

```diff
@@ -0,0 +1,58 @@
+# eq-session in Datastore instead of Redis
+
+Recent scale and stress testing has shown that using Memorystore Redis as the backend for eq-session with the current Runner and Redis architecture does not support the volume of requests required for upcoming Runner use cases.
+
+This test is to investigate the impact of moving eq-session from Redis to Datastore under the current Runner v3.52.0 implementation and web server architecture (gunicorn-async with 7 workers)
+
+https://github.com/ONSdigital/eq-questionnaire-runner/tree/performance-investigation-eq-session-datastore
+
+## Benchmark profile
+
+| Option                 | Value                        |
+| ---------------------- | ---------------------------- |
+| Requests file          | census_household_gb_eng.json |
+| Run time               | 20m                          |
+| User wait time minimum | 1                            |
+| User wait time maximum | 2                            |
+| Clients                | 200                          |
+| Hatch rate             | 200                          |
+| Locust Instances       | 2                            |
+
+## Results
+
+Locust results based on 99th percentile timings.
+
+| Metric | Baseline | Investigation |
+|--------|----------|--------------|
+| Questionnaire GETs | 333ms | 340ms|
+| Questionnaire POSTs | 374ms| 379ms|
+| All requests | 353ms | 359ms |
+| Total requests (over 20 mins) | 750,432 | 746,720 |
```

Comment:
I am not quite sure how with a max usage of up to 16 cores out of 30, the response time is so high. Would be good to know if 100th percentile is particularly high which may have inflated the response times?

Reply:
The 100% should be in the results folder csv files. I will double check and add them if we think it will be useful. We haven't previously.

> I am not quite sure how with a max usage of up to 16 cores out of 30, the response time is so high

Compared to what? We see the response times increase at about ~50% CPU so I don't think that is unexpected. Daily test is < ~50% CPU so it's not unexpected they will see lower latencies in the results. We are comparing the baseline to the investigation here, not to the daily tests. If we want to start doing that then we should run these tests in the eq-test-environment (which I actually suggested for the purposes of this particular investigation but the team decided that wasn't appropriate).

Reply:
We should prob investigate response times at different CPU usages.